dagmc_stats.py

```diff
@@ -300,3 +300,184 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
+        side_lengths.append(np.linalg.norm(my_core.get_coords(vert2)-my_core.get_coords(other_vert)))
+        beta_angles.append(np.arccos((side_lengths[1] * side_lengths[1]
+                            + side_lengths[2] * side_lengths[2]
+                            - side_lengths[0] * side_lengths[0])
+                            /(2.0 * side_lengths[1] * side_lengths[2])))
+    
+
+def get_angles(my_core, tri, vert = None):
+    """
+    Gets the angles of the given triangle
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    tri: triangle
+    vert: the vertex which generates the triangle by get_adj
+
+    outputs
+    -------
+    angles : (list) the angles in the tri
+    """
+    
+    angles = []
+    side_lengths = []
+    coord_list = []
+    index = 3
+    if vert != None:
+        verts = my_core.get_adjacencies(tri, 0, op_type=1)
+        for i in range(len(verts)):
+            if verts[i] == vert:
+                index = i
+        for vert in verts:
+            coords = my_core.get_coords(vert)
+            coord_list.append(coords)
+        side_lengths.append(np.linalg.norm(coord_list[index]-coord_list[index-1]))
+        side_lengths.append(np.linalg.norm(coord_list[index]-coord_list[index-2]))
+        side_lengths.append(np.linalg.norm(coord_list[index-1]-coord_list[index-2]))
+    else:
+        side_lengths = get_tri_side_length(my_core, tri)
+    for side in range(3):
+        angles.append(np.arccos((side_lengths[side] * side_lengths[side]
+                            + side_lengths[side - 2] * side_lengths[side - 2]
+                            - side_lengths[side - 1] * side_lengths[side - 1])
+                            /(2.0 * side_lengths[side] * side_lengths[side - 2])))
+    return angles
+
+
+def gaussian_curvature(my_core, vert):
+    """
+    Gets gaussian curvature of a vert
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    vert : vertex
+
+    outputs
+    -------
+    gc : the gaussian curvature of the vert
+    """ 
+    
+    tris = my_core.get_adjacencies(vert, 2)
+    alpha_angles = 0
+    
+    for tri in tris:
+        alpha_angles += get_angles(my_core, tri, vert = vert)[0]
+    
+    gc = np.abs(2*np.pi - alpha_angles)
+    
+    return gc
+
+
+def get_local_roughness(my_core, vert):
+    """
+    Gets local roughness of a vert
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    vert : vertex
+
+    outputs
+    -------
+    lr : the laplacian value of the vert
+    """ 
+    
+    beta_angles = []
+    d = []
+    gc_j = []
+    sum_d_gc = 0
+    
+    gc_i = gaussian_curvature(my_core, vert)
+    
+    adj_tris = my_core.get_adjacencies(vert, 2, op_type=0)
+    
+    #pair tris
+    pair_tris = []
+    tri_vert = []
+    common_vert = []
+    for tri in adj_tris:
+        tri_vert.append(my_core.get_adjacencies(tri, 0, op_type=1))
+    for tri1 in tri_vert:
```

Comment:
I have a hunch that using sets may make a lot of this simpler...

Reply:
Perhaps we should discuss or study the pyMOAB methods that are designed to do some of this.  Some of what you are doing here is capability built in to MOAB.

Reply:
I would be interested in talking more about how to streamline this because we had a hard time coming up with a method to identify which entities were the relevant ones for calculating the beta angles.

Reply:
This would be a good SW design meeting conversation (cc @bam241)